From 746ac312478efb531dc368d08eeaf496782273ca Mon Sep 17 00:00:00 2001
From: Joel A Fernandes <joelagnel@ti.com>
Date: Wed, 21 Nov 2012 12:19:29 -0600
Subject: [PATCH 6/6] Add capebus override and pinmux for da8xx-dt

Capebus override is left for LT only, rest is kept in common
as there is an HDMI cape as well that could reuse in future.

Signed-off-by: Joel A Fernandes <joelagnel@ti.com>
---
 arch/arm/boot/dts/am335x-bone-common.dtsi |   48 +++++++++++++++++++++++++++++
 arch/arm/boot/dts/am335x-bonelt.dts       |   10 ++++++
 2 files changed, 58 insertions(+)

diff --git a/arch/arm/boot/dts/am335x-bone-common.dtsi b/arch/arm/boot/dts/am335x-bone-common.dtsi
index 46d5f27..4bd1604 100644
--- a/arch/arm/boot/dts/am335x-bone-common.dtsi
+++ b/arch/arm/boot/dts/am335x-bone-common.dtsi
@@ -119,6 +119,32 @@
 			>;
 		};
 
+		nxp_hdmi_cape_pins: nxp_hdmi_cape_pins {
+			pinctrl-single,pins = <
+
+				0xa0 0x08	/* lcd_data0.lcd_data0, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xa4 0x08	/* lcd_data1.lcd_data1, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xa8 0x08	/* lcd_data2.lcd_data2, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xac 0x08	/* lcd_data3.lcd_data3, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xb0 0x08	/* lcd_data4.lcd_data4, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xb4 0x08	/* lcd_data5.lcd_data5, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xb8 0x08	/* lcd_data6.lcd_data6, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xbc 0x08	/* lcd_data7.lcd_data7, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xc0 0x08	/* lcd_data8.lcd_data8, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xc4 0x08	/* lcd_data9.lcd_data9, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xc8 0x08	/* lcd_data10.lcd_data10, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xcc 0x08	/* lcd_data11.lcd_data11, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xd0 0x08	/* lcd_data12.lcd_data12, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xd4 0x08	/* lcd_data13.lcd_data13, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xd8 0x08	/* lcd_data14.lcd_data14, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xdc 0x08	/* lcd_data15.lcd_data15, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT | AM33XX_PULL_DISA */
+				0xe0 0x00	/* lcd_vsync.lcd_vsync, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT */
+				0xe4 0x00	/* lcd_hsync.lcd_hsync, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT */
+				0xe8 0x00	/* lcd_pclk.lcd_pclk, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT */
+				0xec 0x00	/* lcd_ac_bias_en.lcd_ac_bias_en, OMAP_MUX_MODE0 | AM33XX_PIN_OUTPUT */
+			>;
+		};
+
 		bone_geiger_cape_led_pins: pinmux_bone_geiger_cape_led_pins {
 			pinctrl-single,pins = <
 				0xe4 0x07	/* lcd_hsync.gpio2_23, OUTPUT | MODE7 */
@@ -267,6 +293,10 @@
 				compatible = "bone-generic-cape";
 			};
 
+			nxp_hdmi_capebus: cape@6 {
+				compatible = "bone-generic-cape";
+			};
+
 			/* overrides; no EEPROM (prototyping) */
 //			override@3 {
 //				compatible = "bone-capebus-slot-override";
@@ -426,6 +456,24 @@
 	slots = <&cape_eeprom_0 &cape_eeprom_1 &cape_eeprom_2 &cape_eeprom_3>;
 };
 
+/* On-board NXP HDMI Part requires LCDC enabled which is currently
+   only registered using capebus infrastructure */
+&nxp_hdmi_capebus {
+	board-name = "NXP HDMI on capebus";
+	version@00A1 {
+		version = "00A1";
+		dvi {
+			compatible = "da8xx-dt";
+			pinctrl-names = "default";
+			pinctrl-0 = <&nxp_hdmi_cape_pins>;
+
+			ti,hwmods = "lcdc";
+			disp-pll = <135000000>;
+			panel-type = "nxp-720x480@60";
+		};
+	};
+};
+
 &bone_dvi_cape {
 	board-name = "BeagleBone DVI-D CAPE";
 
diff --git a/arch/arm/boot/dts/am335x-bonelt.dts b/arch/arm/boot/dts/am335x-bonelt.dts
index 037594d..44814b4 100644
--- a/arch/arm/boot/dts/am335x-bonelt.dts
+++ b/arch/arm/boot/dts/am335x-bonelt.dts
@@ -37,3 +37,13 @@
 	 	reg = <0x34>;
 	};
 };
+
+&capebus {
+	override@0 {
+		board-name = "NXP HDMI on CapeBus";
+		compatible = "bone-capebus-slot-override";
+		slot = <1>;
+		version = "00A1";
+		manufacturer = "Beagleboardtoys";
+	};
+};
-- 
1.7.9.5

